services/dedup/badger/badger.go

```diff
@@ -66,43 +68,39 @@ func NewBadgerDB(conf *config.Config, stats stats.Stats, path string) *Dedup {
 		path:   path,
 		window: dedupWindow,
 		opts:   badgerOpts,
-		wg:     sync.WaitGroup{},
 		bgCtx:  bgCtx,
 		cancel: cancel,
 	}
 	return &Dedup{
-		badgerDB: db,
-		cache:    make(map[string]bool),
+		badgerDB:    db,
+		uncommitted: make(map[string]struct{}),
 	}
 }
 
-func (d *BadgerDB) Get(key string) (bool, error) {
-	defer d.stats.NewTaggedStat("dedup_get_duration_seconds", stats.TimerType, stats.Tags{"mode": "badger"}).RecordDuration()()
-
-	var found bool
+func (d *BadgerDB) Get(keys []string) (map[string]bool, error) {
+	defer d.stats.NewTaggedStat("dedup_getbatch_duration_seconds", stats.TimerType, stats.Tags{"mode": "badger"}).RecordDuration()()
```

Comment:
We can probably pre-create this stat to avoid calling the constructor each time which does some locking via mutexes. Just saying because it seems that `Get()` might be called a lot.